packages/react-native/Libraries/Blob/__tests__/URL-test.js

```diff
@@ -52,5 +54,75 @@ describe('URL', function () {
     expect(url.pathname).toBe('/docs/path');
     expect(url.port).toBe('8080');
     expect(url.search).toBe('?query=testQuery&key=value');
+
+    // Test searchParams
+    const searchParams = url.searchParams;
+    expect(searchParams.get('query')).toBe('testQuery');
+    expect(searchParams.get('key')).toBe('value');
+
+    const paramsFromString = new URLSearchParams(
+      '?param1=value1+value2&param2=value%20with%20space',
+    );
```

Comment:
nit: Let's be slightly more self-documenting with these cases

```suggestion
    const paramsFromString = new URLSearchParams(
      [
        '?param1=value1',
        '&param2=value2%20with%20spaces',
        '&param3=value3+with+spaces+legacy',
      ].join('')
    );
```